core/thread/thread_unix.odin

```diff
@@ -182,3 +182,65 @@ _terminate :: proc(t: ^Thread, exit_code: int) {
 _yield :: proc() {
 	posix.sched_yield()
 }
+
+_get_name :: proc(thread: ^Thread, allocator: runtime.Allocator, loc: runtime.Source_Code_Location) -> (name:string, err:runtime.Allocator_Error) {
+	// Haiku doesn't have pthread_getname yet
+	when ODIN_OS == .Haiku {
+		unimplemented("core:thread get_name for haiku is not yet supported")
+	}
+
+	tid: posix.pthread_t
+	if thread == nil {
+		tid = posix.pthread_self()
+	} else {
+		tid = thread.unix_thread
+	}
+	
+	buf := make([]u8, _MAX_PTHREAD_NAME_LENGTH, allocator, loc) or_return
+
+	when ODIN_OS == .Darwin || ODIN_OS == .Linux || ODIN_OS == .FreeBSD || ODIN_OS == .NetBSD {
+		pthread_getname_np(tid, raw_data(buf), len(buf))
+	} else when ODIN_OS == .OpenBSD {
+		pthread_get_name_np(tid, raw_data(buf), len(buf))
+	}
+
+	name = transmute(string)buf
```

Comment:
This string will be of length _MAX_PTHRAD_NAME_LENGTH, not the length of the actual name